libutempter/utempter.c

```diff
@@ -69,7 +69,7 @@ validate_hostname(const char *host)
 {
 	for (const char *p = host; p[0]; ++p) {
 		if (!(p == host ? isgraph : isprint)((unsigned char) p[0]))
-			fatal_error("invalid host name");
+			fatal_error("invalid host name (contains 0x%x)", p[0]);
```

Comment:
As the type of p[0] is signed char, I suggest a cast, e.g. ```(unsigned int) (unsigned char) p[0]```,
to avoid potential sign extension when printing it as an integer.

Also, I suggest replacing 0x%x format with %#x.
There is no output difference given that p[0] != 0, but the latter is more idiomatic.

Reply:
Now the line is too long.
Would you mind if I adjust it a bit as in commit  d1a0e4e?

Reply:
No, that's fine; thanks